sdk/transaction-generator/src/lib.rs

```diff
@@ -0,0 +1,334 @@
+        account.address()
+    );
+    for request in txn {
+        diem_retrier::retry_async(diem_retrier::fixed_retry_strategy(5_000, 20), || {
+            let request = request.clone();
+            let c = client.clone();
+            let client_name = format!("{:?}", client);
+            Box::pin(async move {
+                let txn_str = format!("{}::{}", request.sender(), request.sequence_number());
+                debug!("Submitting txn {}", txn_str);
+                let resp = c.submit(&request).await;
+                debug!("txn {} status: {:?}", txn_str, resp);
+
+                resp.map_err(|e| format_err!("[{}] Failed to submit request: {:?}", client_name, e))
+            })
+        })
+        .await?;
+    }
+    let r = wait_for_accounts_sequence(client, slice::from_mut(account))
+        .await
+        .map_err(|_| format_err!("Mint transactions were not committed before expiration"));
+    debug!(
+        "[{:?}] Account {} is at sequence number {} now",
+        client,
+        account.address(),
+        account.sequence_number()
+    );
+    r
+}
+
+async fn wait_for_accounts_sequence(
+    client: &JsonRpcClient,
+    accounts: &mut [LocalAccount],
+) -> Result<(), Vec<(AccountAddress, u64)>> {
+    let deadline = Instant::now() + TXN_MAX_WAIT;
+    let addresses: Vec<_> = accounts.iter().map(|d| d.address()).collect();
+    loop {
+        match query_sequence_numbers(client, &addresses).await {
+            Err(e) => {
+                info!(
+                    "Failed to query ledger info on accounts {:?} for instance {:?} : {:?}",
+                    addresses, client, e
+                );
+                time::sleep(Duration::from_millis(300)).await;
+            }
+            Ok(sequence_numbers) => {
+                if is_sequence_equal(accounts, &sequence_numbers) {
+                    break;
+                }
+                let mut uncommitted = vec![];
+                if Instant::now() > deadline {
+                    for (account, sequence_number) in zip(accounts, &sequence_numbers) {
+                        if account.sequence_number() != *sequence_number {
+                            warn!("Wait deadline exceeded for account {}, expected sequence {}, got from server: {}", account.address(), account.sequence_number(), sequence_number);
+                            uncommitted.push((account.address(), *sequence_number));
+                            *account.sequence_number_mut() = *sequence_number;
+                        }
+                    }
+                    return Err(uncommitted);
+                }
+            }
+        }
+        time::sleep(Duration::from_millis(100)).await;
+    }
+    Ok(())
+}
+
+fn is_sequence_equal(accounts: &[LocalAccount], sequence_numbers: &[u64]) -> bool {
+    for (account, sequence_number) in zip(accounts, sequence_numbers) {
+        if *sequence_number != account.sequence_number() {
+            return false;
+        }
+    }
+    true
```

Comment:
Opinionated nit so feel free to ignore, but what do you think of something like this instead?
```suggestion
      zip(accounts, sequence_numbers).all(|(account, sequence_number)| account.sequence_number() == sequence_number)
```